### DISCUSSION TITLE
Why is pydantic 2 not supported for open api doc generation?

### DISCUSSION THREAD

**Discussion by 555chy** [langchain-ai/langserve#607] [Q&A]
LANGSERVE: ⚠️ Using pydantic 2.6.4. OpenAPI docs for invoke, batch, stream, stream_log endpoints will not be generated. API endpoints and playground should work as expected. If you need to see the docs, you can downgrade to pydantic 1. For example, `pip install pydantic==1.10.13`. See https://github.com/tiangolo/fastapi/issues/10360 for details.

![图片](https://github.com/langchain-ai/langserve/assets/4208102/0a8a1ad1-6693-45b4-8ce1-5583391036c5)

I use pydantic 2.6.4 to run this，but web and doc both not work！

**Comment by eyurtsev** [comment]
@555chy you can support documentation on your own by providing a second implementation for the endpoints and declaring its documentation as a workaround. We're not going to provide an adapter here since it's on FastAPIs road-map and that seems like the easiest place to support it. LangChain itself will be upgrading entirely to pydantic v2 around 2024 July.

**Reply by codekiln** [reply]
> you can support documentation on your own by providing a second implementation for the endpoints and declaring its documentation as a workaround.

@eyurtsev is there an example implementation somewhere? I don't quite understand the idea.
